Cohesity security

yinghuang123 opened the pull request:
Required items, please complete
   
   Change(s):
   - add CohesitySecurity Solution

   Testing Completed:
   - Need Help

   Checked that the validations are passing and have addressed any issues that are present:
   - Need Help

yinghuang123 committed:
initialize the new branch: CohesitySecurity
add the data connector json file.

yinghuang123 committed:
add these files:
    ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.json
    ./Solutions/CohesitySecurity/Playbooks/Incident_Email_Playbook.json
    ./Solutions/CohesitySecurity/Playbooks/Incident_VM_Playbook.json
need to find a azure way to convert the RansomwareAlert.json to yaml format,

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
add these files:
    ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.yaml
    ./Solutions/CohesitySecurity/SolutionMetadata.json
    ./Tools/Create-Azure-Sentinel-Solution/V2/input/Solution_CohesitySecurity.json
    ./Tools/Create-Azure-Sentinel-Solution/input/Solution_CohesitySecurity.json
    ./Tools/Create-Sentinel-Solution/V2/input/Solution_CohesitySecurity.json

I feel we should use the v2 build script
    ./Tools/Create-Azure-Sentinel-Solution/V2/createSolutionV2.ps1
instead of
    ./Tools/Create-Azure-Sentinel-Solution/createSolution.ps1

It failed on other Solutions, dued to some hard-coded path, for example:
    Downloading C:\GitHub\azure\Solutions\CiscoDuoSecurity/Workbooks/CiscoDuo.json
    Failed to download C:\GitHub\azure\Solutions\CiscoDuoSecurity/Workbooks/CiscoDuo.json -- Please ensure that it exists in C:%5CGitHub%5Cazure%5CSolutions%5CCiscoDuoSecurity/

For the reason mentioned above, just temporarily delete these files, in order to have a pass build:
    ./Tools/Create-Azure-Sentinel-Solution/V2/input/Solution_CiscoDuoSecurity.json
    ./Tools/Create-Azure-Sentinel-Solution/input/Solution_CiscoWSA.json
    ./Tools/Create-Azure-Sentinel-Solution/input/Solution_SIGNL4.json
    ./Tools/Create-Azure-Sentinel-Solution/input/Solution_ZeroNetworks.json
in our final version, need to get back these files.

what's next:
    These 3 files are duplicate, need to remove 2, and only keep one in our final version.
    for I am not sure, and its doc and examples are conflicting with each other, again,,so, I just keep 3 in different location:
        ./Tools/Create-Azure-Sentinel-Solution/V2/input/Solution_CohesitySecurity.json
        ./Tools/Create-Azure-Sentinel-Solution/input/Solution_CohesitySecurity.json
        ./Tools/Create-Sentinel-Solution/V2/input/Solution_CohesitySecurity.json
    write another script to only build our Solution, so we could get back those temporarily deleted files
    Fix the last failures:
        Failed arm-ttk (Test-AzTemplate): Package
        Failed arm-ttk (Test-AzTemplate) on solutions: Package

Will try again after dinner.

yinghuang123 committed:
clean some un-needed files
update the playbook *.json files
so, now build, and auto validataion pass

yinghuang123 committed:
get back the other Solutions under input folder, to minimize the diff to mastter
add the yaml format of Analytic Rules

yinghuang123 committed:
add stuff for v1 createsolution script
manually change some version value to pass build validation for v1 script
still resulted in the same error with v2, during deployment

eerus committed:
Update SolutionMetadata.json

"cohesity security" -> "cohesitysecurity"
Added "verticals": []

eerus committed:
Update Solution_CohesitySecurity.json

        "Analytic Rules/RansomwareAlert.json" ->         "Analytic Rules/RansomwareAlert.yaml"

eerus committed:
Update Solution_CohesitySecurity.json

"Analytic Rules/RansomwareAlert.yaml" - > "Analytic Rules/Ransomware Alert.yaml"

eerus committed:
Update Solution_CohesitySecurity.json

Ransomware Alert.yaml -> RansomwareAlert.yaml

yinghuang123 committed:
+++ ./Solutions/CohesitySecurity/readme
    instruct how to build, validate, etc:
+++ ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.yaml
    manually update thise generated yaml file to pass the build validatation
+++ ./Solutions/CohesitySecurity/SolutionMetadata.json
    update the correct tier value

yinghuang123 committed:
+++ ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.yaml
    tried automated process to generate this file

ishitagupta25 commented on Solutions/CohesitySecurity/Data Connectors/helios_connector.json:
```diff
@@ -0,0 +1,117 @@
+                    "APIKeyName": "apiKey"
+                },
+                "request": {
+                    "apiEndpoint": "https://helios.cohesity.com/mcm/alerts?maxAlerts=10&alertCategoryList=kSecurity&alertStateList=kOpen",
```
For our internal testing, apiEndpoint URL should be: helios-test2.cohesitycloud.co 
For packaging and publication: helios.cohesity.com is ok.

yinghuang123 replied:
so, just keep this unchanged right now.

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/Alert_VM_Playbook/Alert_VM_Playbook.json:
```diff
@@ -0,0 +1,111 @@
+{
```
Can remove this playbook. It was created for understanding and reference purpose only.

yinghuang123 replied:
this one is just removed.

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/Close-SentinelIncident-from-SNOW/Close-SentinelIncident-from-SNOW.json:
```diff
@@ -0,0 +1,390 @@
+{
```
Remove this playbook. We have marked this item for future work.

yinghuang123 replied:
removed.

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/CreateSNOWRecord/CreateSNOWRecord.json:
```diff
@@ -0,0 +1,215 @@
+{
```
Remove this playbook. Again for reference purpose only.

yinghuang123 replied:
removed.

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/Cohesity_Alert_Email_Playbook/Cohesity_Alert_Email_Playbook.json:
```diff
@@ -0,0 +1,197 @@
+{
```
Remove this playbook.

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/Incident_Email_Playbook/Incident_Email_Playbook.json:
```diff
@@ -0,0 +1,199 @@
+                                "variables": [{
+                                    "name": "EmailId",
+                                    "type": "string",
+                                    "value": "sentinel-dl@cohesity.com"
```
Sanitise Email ID.

yinghuang123 replied:
done in latest commit

ishitagupta25 commented on Solutions/CohesitySecurity/Playbooks/Incident_VM_Playbook/Incident_VM_Playbook.json:
```diff
@@ -0,0 +1,183 @@
+{
```
Please update this playbook from yings-instance. Have added more fields in HTTP request. It won't work otherwise.

yinghuang123 replied:
just update this.

ishitagupta25 commented on Tools/Create-Azure-Sentinel-Solution/input/Solution_CohesitySecurity.json:
```diff
@@ -0,0 +1,25 @@
+        "Analytic Rules/RansomwareAlert.yaml"
+    ],
+    "Playbooks": [
+        "Playbooks/Alert_VM_Playbook/Alert_VM_Playbook.json",
```
Remove extra playbook's name from here.

yinghuang123 replied:
updated in latest commit

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
+++ ./Solutions/CiscoDuoSecurity/Package/mainTemplate.json
+++ ./Solutions/CiscoWSA/Package/mainTemplate.json
+++ ./Solutions/SIGNL4/Package/mainTemplate.json
+++ ./Tools/Create-Azure-Sentinel-Solution/V2/input/Solution_CiscoDuoSecurity.json
+++ ./Tools/Create-Azure-Sentinel-Solution/input/Solution_CiscoWSA.json
+++ ./Tools/Create-Azure-Sentinel-Solution/input/Solution_SIGNL4.json
+++ ./Tools/Create-Azure-Sentinel-Solution/input/Solution_ZeroNetworks.json
    add back these files to minimize the diff to master branch.

remove some un-used playbooks.
add place holders, so as to sanitize and customize PII value.
update readme, to illustrate how to use build.sh to build the package.

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
small update to Incident_VM_Playbook.json

yinghuang123 committed:
small update to readme

yinghuang123 committed:
+++ ./Solutions/CohesitySecurity/Playbooks/SNOW-CreateAndUpdateIncident/SNOW-CreateAndUpdateIncident.json
    fix a weird validation issue
+++ ./Solutions/CohesitySecurity/build.sh
+++ ./Tools/Create-Azure-Sentinel-Solution/build_cohesity.ps1
    add some build files for CohesitySecurity

yinghuang123 committed:
--- ./Solutions/CohesitySecurity/build.sh
--- ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.json
+++ ./Solutions/CohesitySecurity/Playbooks/SNOW-CreateAndUpdateIncident/SNOW-CreateAndUpdateIncident.json
+++ ./Solutions/CohesitySecurity/build.ps1
rename from Tools/Create-Azure-Sentinel-Solution/build_cohesity.ps1
rename to Solutions/CohesitySecurity/build_one_solution.ps1

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
update readme file.

yinghuang123 committed:
fix the Playbooks/SNOW-CreateAndUpdateIncident
1: remove "-" in variables name.
    avoid using - in variable name of Playbooks.
2: change from Microsoftsentinel to Azuresentinel in api.id
    maybe this Playbook is based on an old template?

yinghuang123 committed:
update the "publisherId" and "offerId"

yinghuang123 committed:
Fix Azure Pipelines validation error.

yinghuang123 committed:
+++ ./.script/tests/KqlvalidationsTests/CustomTables/HeliosAuditNativePoller_CL.json

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
small update to Analytic rule and helios_connector.

+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger.sln
+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger/AlertHttpTrigger.cs
+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger/AlertHttpTrigger.csproj
+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger/deploy.sh
+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger/host.json
+++ ./Solutions/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger/local.settings.json

yinghuang123 committed:
Merge branch 'Azure:master' into CohesitySecurity

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/master' into CohesitySecurity

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/CohesitySecurity' into CohesitySecurity

yinghuang123 committed:
update the queryFrequency and queryPeriod

yinghuang123 committed:
update metadata for playbooks.

yinghuang123 committed:
parameterize the CohesitySecurity/AlertHttpTrigger
add resetRedis to reset redis key value
remove ﻿ in *.json files.

yinghuang123 committed:
+++ ./Sample Data/Custom/HeliosAuditNativePoller_CL.json

yinghuang123 committed:
add try catch in DataConnectors/CohesitySecurity/AlertHttpTrigger/AlertHttpTrigger.cs

yinghuang123 committed:
replace the AlertHttpTrigger with Helios2Sentinel data connector.
add README.md to all the playbooks.
rename all the playbooks to azuredeploy.json file as suggested.

eerus committed:
Update readme.md

Typo

yinghuang123 committed:
--- ./Solutions/CohesitySecurity/Data Connectors/helios_connector.json

eerus commented:
Hi Microsoft team!
We don't understand how to fix [Analyze (csharp)] test as it reports issues in the code that we don't own and the previous commits didn't result in this failure. Here's the exact message 

> We were unable to automatically build your code. Please replace the call to the autobuild action with your custom build steps.  Failure invoking /opt/hostedtoolcache/CodeQL/0.0.0-20221123/x64/codeql/csharp/tools/autobuild.sh with arguments

Further reading didn't give more clarifications

>   /home/runner/.nuget/packages/microsoft.net.sdk.functions/3.0.3/build/Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : You must install or update .NET to run this application. [/home/runner/work/Azure-Sentinel/Azure-Sentinel/DataConnectors/O365 DataCSharp/Teams.CustomConnector.Serverless/Teams.CustomConnector.Serverless.csproj]

What do we need to do? We'd appreciate any help or an explanation.

yinghuang123 committed:
Squashed commit of the following:

commit df0261d7ff646c304381b10595cd66d6dbf501e9
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 14:39:03 2022 -0800

    fix typo.

commit 3cbc054f560096ff6a152ac8c4cfe6c86c993795
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 14:22:53 2022 -0800

    update the host.json for queue policy.

commit 5c975aa3bd8045e8e913b4cf2f86dddd0035f226
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 14:11:29 2022 -0800

    delete these 2 not in use files.

commit 34d4b4ea72e6055cbe0724b196ee87cb61bd5de0
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 14:01:47 2022 -0800

    avoid recursion in catch.
    move FixedDelayRetry into host.json file.

commit a5675c32eca3a8b7006585e4985403430e28a67e
Merge: e4316deb4 24fdabb0d
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 13:43:06 2022 -0800

    Merge branch 'CohesitySecurity.internal' of ../../BareReps/workspace/Azure-Sentinel1 into CohesitySecurity.internal

commit e4316deb49a3514edbd112305ced244f37ce4aad
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 13:43:06 2022 -0800

    fix the indent issue.

commit 24fdabb0d5ce75cb4f6c587ff4df6ccdf2331702
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Mon Dec 12 13:39:58 2022 -0800

    n

commit 91a8b155d839783e4b3977ced1738f9916fc4edb
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Mon Dec 12 10:19:21 2022 -0600

    Update readme.md

    Formatting changes

commit eba35f847f4759335c94d8776169856effe1bdf2
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Mon Dec 12 10:18:44 2022 -0600

    Update readme.md

    Formatting changes

commit 4922249d7c41619c0ea86c1dd7e8bc93228b8c8b
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Mon Dec 12 10:17:32 2022 -0600

    Update readme.md

    Updated links to individual deployment instructions.

commit d0504d30064fef632d91cdac07d8cded6941a787
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Sat Dec 10 22:47:59 2022 -0800

    add lock to protect the queue in multi-thread scenarios.

commit 79216fa51eaccf5ed0fe6b460cc00ca9f43ff9f2
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Fri Dec 9 16:27:40 2022 -0800

    update readme file to add introduction about CohesityQueueName.

commit 3684302ebd5cd47a4df31adcef34f984182eacf6
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Fri Dec 9 16:07:24 2022 -0800

    moving the try catch logic up.
    Add FixedDelayRetry for QueueTrigger.

commit 2bb8bdb2dff613feaf76bf76c91b124ad40aec29
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Fri Dec 9 09:46:22 2022 -0600

    Update Solution_CohesitySecurity.json

    Changed per Microsoft review

commit 2e4cd092da35eac529ad40cf13965891bcd23e7a
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Fri Dec 9 09:40:25 2022 -0600

    Update SolutionMetadata.json

    Changed per Microsoft code review

commit 4f9275a6c72fb1a3c43e954187aa78b03ef7933e
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 20:30:33 2022 -0600

    Update readme.md

    Adding a testing step

commit 0e6eef68994005f3910d9ea1f19316568ed605f4
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 20:22:54 2022 -0600

    Update readme.md

    Added more testing instructions

commit 868e8aff161f4545303844273885bfaf80f1f850
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 20:17:29 2022 -0600

    Update SolutionMetadata.json

    Formatting changes

commit 2681e4e2104e1b3bb9fdcc2a306396402a55c2a5
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 20:16:47 2022 -0600

    Update SolutionMetadata.json

    Expanding categories and updating support info

commit 456cde77cf36f57505d799ae641e4b0860801837
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 20:11:55 2022 -0600

    Update readme.md

    Polished text

commit e7596e2376153ff6a5a9e304f78d7ef44b6566a0
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 19:51:46 2022 -0600

    Update Solution_CohesitySecurity.json

    Added product description

commit 3b0d6cae138980f4c2b0689e743a14729ce8f5cd
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Thu Dec 8 17:49:44 2022 -0800

    obfuscate the CohesityQueueName
    multi-thread improve ParseAlertToQueue function

commit 6a010359f4fc1bf4452b0ad82f473bb89115e4e2
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 17:34:47 2022 -0600

    Create readme.md

    Final version

commit f062e826cc2f7aad7c71303f74b5343c6b7ba0de
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 16:39:52 2022 -0600

    Update readme.md

    Fixed typos

commit 16dd47d262e6e8af7bd0154049f6516be88ecf03
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 16:32:18 2022 -0600

    Update readme.md

    Complete version

commit 1cc3573c4652b5b37fcbcd390d60290d83390991
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 13:45:14 2022 -0600

    Update readme.md

    Typo

commit 882e06b81c24a3f066362e25d163c03727476f8b
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 13:09:45 2022 -0600

    Create readme.md

    Initial commit

commit a3604698a1cbadefb5c4eab32160ef378bb93eb6
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:55:06 2022 -0600

    Create readme.md

    Initial version

commit a3691ffe3cd4c566dc7ff19fbf25890367e4c893
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:27:32 2022 -0600

    Update readme.md

    Removed unnecessary parameter

commit b54d3775058650cfdcd55b5f53b32eb183e1a4cc
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:26:47 2022 -0600

    Update readme.md

    Formatting changes

commit 7a1d5c44c6b3928a8c19d09b29516f4bd9e03c3f
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:22:17 2022 -0600

    Update readme.md

    Corrected typos

commit beba853c334afcc41c41b9cba573a0017114054f
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:05:11 2022 -0600

    Update readme.md

    Corrected the URI

commit f57caf6c0e43e3c367c756e53c686919bdb33a31
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 12:03:07 2022 -0600

    Update readme.md

    Initial version

commit e6456173f4d0cde1d28279c995fca17f4210764e
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 11:45:53 2022 -0600

    Update readme.md

    Added formatting

commit 776e8f5f9f77360dc5944bb0ea42683e66244c57
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 11:45:12 2022 -0600

    Update readme.md

    Completed the description

commit 9f32298f3f806649e3f7e892ba3b8d38f163d9bb
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 11:12:00 2022 -0600

    Update readme.md

    Initial version

commit 124d09ab973d57965a14011be1acd5d05feeedef
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Thu Dec 8 10:57:08 2022 -0600

    Update readme.md

    Added word Cohesity to the playbook name

commit 72ffa6e79eb3f3bee7b6603bc576222d59701120
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 19:55:55 2022 -0600

    Update readme.md

    Formatting changes

commit 63d91a5d6a0351db688cf5442d39c78dcaf76440
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 19:48:38 2022 -0600

    Update readme.md

    Updated ARM template

commit 9778263e071c5813200c46f27fffaebfbcefbd08
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 19:37:38 2022 -0600

    Update readme.md

    Trying new URI format

commit 30630efe42cf7ef46f5af74d7042c5db5f851389
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 19:30:39 2022 -0600

    Update readme.md

    Changed URI again

commit 0622eba625495e163e2b36a249a550728100ab42
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 19:29:04 2022 -0600

    Update readme.md

    Changed URI for ARM template

commit 6b33b2525f6df624af25f80762463876ddfdd5d5
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Wed Dec 7 17:15:05 2022 -0800

    obfuscate the values in local.settings.json files.
    rename cohesity.config to cohesity.json, and update the build.ps1, readme.md.

    The github/codeql-action/autobuild keeps failing, but DataConnectors/CohesitySecurity/Helios2Sentinel build succeeded, as shown in following log.
    thus will disable the autobuild work flow.
    =============================
    MSBuild version 17.4.0+18d5aef85 for .NET
      Determining projects to restore...
      Restored /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.csproj (in 241 ms).
      Restored /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.csproj (in 241 ms).
    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.cs(65,34): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.csproj]
      IncidentProducer -> /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/bin/Debug/net6.0/Helios2Sentinel.dll
    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs(30,32): warning CS0618: 'AuthenticationContext.AcquireTokenAsync(string, ClientCredential)' is obsolete: 'Microsoft.IdentityModel.Clients.ActiveDirectory is deprecated and replaced with Microsoft.Identity.Client. To migrate a daemon application, or service to service flow, see https://aka.ms/adal-to-msal-net/daemon' [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.csproj]
    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs(39,51): warning SYSLIB0014: 'WebRequest.Create(Uri)' is obsolete: 'WebRequest, HttpWebRequest, ServicePoint, and WebClient are obsolete. Use HttpClient instead.' [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.csproj]
      IncidentConsumer -> /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/bin/Debug/net6.0/IncidentConsumer.dll

    Build succeeded.

    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.cs(65,34): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentProducer/IncidentProducer.csproj]
    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs(30,32): warning CS0618: 'AuthenticationContext.AcquireTokenAsync(string, ClientCredential)' is obsolete: 'Microsoft.IdentityModel.Clients.ActiveDirectory is deprecated and replaced with Microsoft.Identity.Client. To migrate a daemon application, or service to service flow, see https://aka.ms/adal-to-msal-net/daemon' [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.csproj]
    /Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.cs(39,51): warning SYSLIB0014: 'WebRequest.Create(Uri)' is obsolete: 'WebRequest, HttpWebRequest, ServicePoint, and WebClient are obsolete. Use HttpClient instead.' [/Users/ying.huang/workspace/Azure-Sentinel/DataConnectors/CohesitySecurity/Helios2Sentinel/IncidentConsumer/IncidentConsumer.csproj]
        3 Warning(s)
        0 Error(s)

    Time Elapsed 00:00:03.65

commit 48a92fa01538c5f9682cf85ec863fee7034db2bc
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:56:38 2022 -0600

    Update readme.md

commit fff7b50eef941fc31baaf0132fabc2a5c8be4580
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:55:03 2022 -0600

    Update readme.md

    URI test

commit d9e31f7740e6848b76bb8fb52b23aea4d447717d
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:53:17 2022 -0600

    Update readme.md

    Another link test

commit d2c11aa620bfc8f677de973a4781997349e9573c
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:49:15 2022 -0600

    Update readme.md

    Another URI test

commit 4af3e3078bc46faae5f25ae60db23add4dc64f1e
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:48:19 2022 -0600

    Update readme.md

    Trying different link

commit 3137eb24dbd9c65b11d9893f15b749bfacebc503
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:40:02 2022 -0600

    Update readme.md

    Changed URI

commit 2483ea6d37e3073fd9af84ece13a9bfacf0efa0b
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Wed Dec 7 13:31:11 2022 -0800

    --- ./Solutions/CohesitySecurity/Analytic Rules/RansomwareAlert.yaml
    --- ./Solutions/CohesitySecurity/Data Connectors/helios_connector.json
    +++ ./Solutions/CohesitySecurity/Package/createUiDefinition.json
    +++ ./Solutions/CohesitySecurity/Package/mainTemplate.json
    update the build script.

commit fefdf8cfd2bedde33c8f9e866653cee2a2fd5ee0
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:25:13 2022 -0600

    Update readme.md

    Testing ARM

commit 19ec494e83bb4ddbf7f987e89d90597282fdf247
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:18:22 2022 -0600

    Update readme.md

    Corrected playbook description

commit 59cd9b5250fb8fe40cf8cba981f426562aadcd90
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:04:28 2022 -0600

    Update readme.md

    Added TBD for package directory

commit 66bdab4f47ce3a10b3f9520c8a8a5302a9dc3c7c
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:02:15 2022 -0600

    Update readme.md

    Fixed typos

commit cd7ec9fd1da7147aaa9adf3bd9308a44a877ad1f
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 15:00:08 2022 -0600

    Update readme.md

    Minor fixes

commit 7ba845641f2ee7eef676988e82f53c221bb4e28a
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:45:22 2022 -0600

    Update readme.md

    Deployment steps corrections

commit e6dd9c5fce55658a1ef567f8666252421c922828
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:42:44 2022 -0600

    Update readme.md

commit 29ca436444bcc5cb2da9df46790533969402d4d5
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:41:33 2022 -0600

    Update readme.md

    Setting ref line

commit b1301e2b6ee1513f5c7d6c7fdd02f071279773fa
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:40:34 2022 -0600

    Update readme.md

    Fixing URL again

commit 6d7012c1b57ab0232a9963345d2b68d55b1b4639
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:39:11 2022 -0600

    Update readme.md

    Fixing build instructions URL

commit 098e27bbf719ed680bbbec1da5698d8fe7e4bbca
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:36:40 2022 -0600

    Update readme.md

    Escaped URL

commit 61ea1d567270fb2c0056ed766771050165898d27
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:35:11 2022 -0600

    Update readme.md

    Fixed step #1

commit 0d071a0b695060caffdf0a28f00d2dae927fb5ca
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:34:04 2022 -0600

    Update readme.md

    Typo

commit 028950c162cd7c3fbe21d3f4daeaa980da6be8ca
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:33:26 2022 -0600

    Update readme.md

    Fixed link for the package directory

commit 15af0de66deecf90fc0d631300cd031c20174090
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 13:31:25 2022 -0600

    Update readme.md

    Added references to other deployment instructions

commit f7e5af2070474d57a0233b238a864e3c37d422dc
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Wed Dec 7 11:30:56 2022 -0600

    Update readme.md

    First draft of deployment instructions

commit b83e93aea069b80ff535acaa69149e37f900e3da
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Tue Dec 6 19:29:02 2022 -0600

    Update readme.md

commit 372ae3e17ed3e21ec5958a6619071d1a9db96564
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Tue Dec 6 19:26:25 2022 -0600

    Update readme.md

commit e6b15a5793fe2736c42293c3d2d3d7d8af140cd9
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Tue Dec 6 19:18:30 2022 -0600

    Update readme.md

    Updated link to Azure deployment

commit 3447201611a96ce8b4989483c385877b798016a2
Author: eerus <115032728+eerus@users.noreply.github.com>
Date:   Tue Dec 6 18:56:16 2022 -0600

    Update readme.md

    Added link to the playbook template

commit 36c776fd5b20c27ab9b6668c4c12d3ff89621de5
Author: Ying Huang <ying.huang@cohesity.com>
Date:   Fri Dec 2 15:54:33 2022 -0800

    Just initialize this new branch for Cohesity internal collaboration/review.

yinghuang123 committed:
delete those files not in use.

yinghuang123 committed:
remove en-us in the link, to pass buildbot validation.

yinghuang123 committed:
rename playbook from Incident_Email_Playbook to Cohesity_Send_Incident_Email
remove alerts loop in that playbook.

yinghuang123 committed:
update playbook: Cohesity_Send_Incident_Email
